### ISSUE TITLE
possibly null value provided when inherit constant and after null check

### ISSUE THREAD

**Issue by xpader** [vimeo/psalm#10058]
In example `const FOO` is mark as `string|null`, and override in subclass World, and checked `static::FOO !== null` but psalm still think it's `possibly null value provided`.

https://psalm.dev/r/e20dea009b

**Comment by orklah** [comment]
Yeah, that's a current limitation in Psalm, we don't restrict the type of a constant based on checks made on it. (We could but it creates a very big performance issue on switch expression)

**Comment by weirdan** [comment]
To start with, we ignore `@var` on constants: https://psalm.dev/r/fca65947ec

**Comment by xpader** [comment]
I think the point is const in class can be override by subclass.
